src/components/questions-test/QuestionsTest.tsx

```diff
@@ -1,50 +1,41 @@
 import * as React from "react";
 
-import { IQuestionCardAnswer } from "./question-card/question-card-answer/QuestionCardAnswer";
-import QuestionCard, { IQuestionCard } from "./question-card/QuestionCard";
+import { Answer } from "./question-card/question-card-answer/QuestionCardAnswer";
+import QuestionCard, { Question } from "./question-card/QuestionCard";
 
 import "./QuestionsTest.scss";
 
-interface IQuestionTestProps {
+interface QuestionTestProps {
   title: string;
-  questions: IQuestionCard[];
-  onQuestionAnswer?: (
-    question: IQuestionCard,
-    answer: IQuestionCardAnswer
-  ) => void;
+  questions: Question[];
+  onQuestionAnswer?: (question: Question, answer: Answer) => void;
 }
 
-class QuestionsTest extends React.Component<IQuestionTestProps> {
-  public render() {
-    const { questions, title } = this.props;
-    const hasQuestions = !!questions.length;
-
-    return (
-      <div className="questions-test">
-        {hasQuestions && (
-          <QuestionCard
-            title={title}
-            question={questions[0]}
-            onAnswerClick={this.onAnswerClickListener(questions[0])}
-          />
-        )}
-      </div>
-    );
-  }
-
-  private onAnswerClickListener(question: IQuestionCard) {
-    return (answer: IQuestionCardAnswer) =>
-      this.onQuestionAnswerClick(question, answer);
-  }
-
-  private onQuestionAnswerClick(
-    question: IQuestionCard,
-    answer: IQuestionCardAnswer
-  ) {
-    if (this.props.onQuestionAnswer) {
-      this.props.onQuestionAnswer(question, answer);
-    }
-  }
+function QuestionsTest({
+  questions,
+  title,
+  onQuestionAnswer,
+}: QuestionTestProps) {
+  const onAnswerClickListener = React.useCallback(
+    (question: Question) => {
+      return (answer: Answer) => onQuestionAnswer?.(question, answer);
+    },
+    [onQuestionAnswer]
+  );
+
+  const hasQuestions = !!questions.length;
+
+  return (
+    <div className="questions-test">
+      {hasQuestions && (
```

Comment:
Is there a reason to not just write `!!questions.length` here?

Reply:
Mostly because it was like that before, but on reflection it's a more descriptive name for what the check means (self-documenting code).